ValveResourceFormat/IO/GltfModelExporter.cs

```diff
@@ -44,29 +44,36 @@ public class GltfModelExporter
         public bool AdaptTextures { get; set; } = true;
 
         private string DstDir;
-        private readonly IDictionary<string, Node> LoadedUnskinnedMeshDictionary = new Dictionary<string, Node>();
+        private readonly IDictionary<string, Mesh> LoadedMeshDictionary = new Dictionary<string, Mesh>();
 
         public static bool CanExport(Resource resource)
             => ResourceTypesThatAreGltfExportable.Contains(resource.ResourceType);
 
         public void Export(Resource resource, string targetPath)
         {
-            switch (resource.ResourceType)
+            try
             {
-                case ResourceType.Mesh:
-                    ExportToFile(resource.FileName, targetPath, new VMesh(resource));
-                    break;
-                case ResourceType.Model:
-                    ExportToFile(resource.FileName, targetPath, (VModel)resource.DataBlock);
-                    break;
-                case ResourceType.WorldNode:
-                    ExportToFile(resource.FileName, targetPath, (VWorldNode)resource.DataBlock);
-                    break;
-                case ResourceType.World:
-                    ExportToFile(resource.FileName, targetPath, (VWorld)resource.DataBlock);
-                    break;
-                default:
-                    throw new ArgumentException($"{resource.ResourceType} not supported for gltf export");
+                switch (resource.ResourceType)
+                {
+                    case ResourceType.Mesh:
+                        ExportToFile(resource.FileName, targetPath, new VMesh(resource));
+                        break;
+                    case ResourceType.Model:
+                        ExportToFile(resource.FileName, targetPath, (VModel)resource.DataBlock);
+                        break;
+                    case ResourceType.WorldNode:
+                        ExportToFile(resource.FileName, targetPath, (VWorldNode)resource.DataBlock);
+                        break;
+                    case ResourceType.World:
+                        ExportToFile(resource.FileName, targetPath, (VWorld)resource.DataBlock);
+                        break;
+                    default:
+                        throw new ArgumentException($"{resource.ResourceType} not supported for gltf export");
+                }
+            }
+            finally
+            {
+                LoadedMeshDictionary.Clear();
```

Comment:
Other ExportToFile overloads are public too, and it won't be cleared if consumer calls them directly instead of this method.

Reply:
Well then the only adequate solution left is passing cache as argument. Done.